### PR TITLE
[gosrc2cpg] - support to exclude files

### PR DESCRIPTION
**Pull request by ankit-privado**
we can now exclude file for go frontend:

`importCode.golang.apply(inputPath = "path", projectName = "name", args = List("--exclude-regex", "regex"))`

### CODE REVIEW

**Review by DavidBakerEffendi — APPROVED** [review]

File: joern-cli/frontends/gosrc2cpg/src/main/scala/io/joern/gosrc2cpg/utils/AstGenRunner.scala [hunk 1]
```diff
@@ -134,13 +134,15 @@ class AstGenRunner(config: Config) {
     }
   }
 
-  private def runAstGenNative(in: String, out: File): Try[Seq[String]] =
-    ExternalCommand.run(s"$astGenCommand -out ${out.toString()} $in", ".")
+  private def runAstGenNative(in: String, out: File, exclude: String): Try[Seq[String]] = {
+    val excludeCommand = if (exclude.isEmpty) "\"\"" else s"\"$exclude\""
```
Comment (pandurangpatil) [inline joern-cli/frontends/gosrc2cpg/src/main/scala/io/joern/gosrc2cpg/utils/AstGenRunner.scala:138]: @ankit-privado Can you try following? 

```
val excludeCommand = if (exclude.isEmpty) "" else s"-exclude $exclude"
ExternalCommand.run(s"$astGenCommand $excludeCommand -out ${out.toString()} $in", ".")

```

**Review by pandurangpatil — COMMENTED** [review]
Can you try the suggestion?

File: joern-cli/frontends/gosrc2cpg/src/main/scala/io/joern/gosrc2cpg/utils/AstGenRunner.scala (hunk 1 above)
Comment (ankit-privado) [reply joern-cli/frontends/gosrc2cpg/src/main/scala/io/joern/gosrc2cpg/utils/AstGenRunner.scala:138]: initially I was also doing this, but this is command line string. if we do this then command look like `..-exclude -out..` but we need `..-exclude "" -out..`

@pandurangpatil

Comment (pandurangpatil) [reply joern-cli/frontends/gosrc2cpg/src/main/scala/io/joern/gosrc2cpg/utils/AstGenRunner.scala:138]: What I am suggesting should generate `.. -out ..`

Comment (pandurangpatil) [reply joern-cli/frontends/gosrc2cpg/src/main/scala/io/joern/gosrc2cpg/utils/AstGenRunner.scala:138]: If `goastgen` always expects a value to be passed for `-exclude`, then I suppose we need to fix that on `goastgen`

Comment (ankit-privado) [reply joern-cli/frontends/gosrc2cpg/src/main/scala/io/joern/gosrc2cpg/utils/AstGenRunner.scala:138]: okay okay.

**Pull request merged**